diff -Naur kde1-kdenetwork-1029167fdfe0bb9e0d49cb3508b86e332ccebaaf.orig/kmail/CMakeLists.txt kde1-kdenetwork-1029167fdfe0bb9e0d49cb3508b86e332ccebaaf/kmail/CMakeLists.txt
--- kde1-kdenetwork-1029167fdfe0bb9e0d49cb3508b86e332ccebaaf.orig/kmail/CMakeLists.txt	2017-02-07 02:15:37.000000000 +1300
+++ kde1-kdenetwork-1029167fdfe0bb9e0d49cb3508b86e332ccebaaf/kmail/CMakeLists.txt	2017-10-08 18:04:14.013356001 +1300
@@ -3,6 +3,7 @@
 include_directories(
     ${QT_INCLUDE_DIR}
     ${KDE1_INCLUDE_DIR}
+    ../mimelib
 )
 
 set(KMAIL_SRCS
