diff --git a/Makefile b/Makefile
index d2f588cc..4dcf5005 100644
--- a/Makefile
+++ b/Makefile
@@ -882,7 +882,6 @@ $(BIN): $(OBJ)
 	$V$(CXX) -o $@ $(LINKFLAGS) $(OBJ) $(LIBS)
 	objcopy --only-keep-debug $(BIN) $(BIN).debug
 	objcopy --add-gnu-debuglink=$(BIN).debug $(BIN)
-	strip $(BIN)
 
 $(OBJDIR)%.o: src/%.cpp
 	@mkdir -p $(dir $@)
diff --git a/src/mir/check_full.cpp b/src/mir/check_full.cpp
index 1f86c40a..6cf191f6 100644
--- a/src/mir/check_full.cpp
+++ b/src/mir/check_full.cpp
@@ -381,8 +381,8 @@ namespace
                 void mark_from_state(const ValueStates& vss, const State& s) {
                     if(s.is_composite()) {
                         used.at(s.index-1) = true;
-                        for(const auto& s : vss.inner_states.at(s.index-1))
-                            mark_from_state(vss, s);
+                        for(const auto& s2 : vss.inner_states.at(s.index-1))
+                            mark_from_state(vss, s2);
 
                         // TODO: Should this compact composites with all-equal inner states?
                     }
