diff --git a/Java/src/com/jameslow/AirPlay.java b/Java/src/com/jameslow/AirPlay.java
index d1cbd31..c675473 100755
--- a/Java/src/com/jameslow/AirPlay.java
+++ b/Java/src/com/jameslow/AirPlay.java
@@ -378,13 +378,14 @@ public class AirPlay {
 		}
 	}
 	protected static Service[] formatSearch(ServiceInfo[] services) throws IOException {
-		Service[] results = new Service[services.length];
+		java.util.List<Service> results = new java.util.ArrayList(services.length);
 		for (int i = 0; i < services.length; i++) {
 			ServiceInfo service = services[i];
 			Inet4Address[] addresses = service.getInet4Addresses();
-			results[i] = new Service(addresses[0].getHostAddress(), service.getPort(), service.getName());
+                       if (addresses.length > 0)
+			    results.add(new Service(addresses[0].getHostAddress(), service.getPort(), service.getName()));
 		}
-		return results;
+		return results.toArray(new Service[results.size()]);
 	}
 	public static java.util.List<Service> search() throws IOException {
 		return search(1000);
