diff --git a/Java/.project b/Java/.project
deleted file mode 100644
index 8d56b90..0000000
--- a/Java/.project
+++ /dev/null
@@ -1,11 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<projectDescription>
-	<name>OpenAirPlay</name>
-	<comment></comment>
-	<projects>
-	</projects>
-	<buildSpec>
-	</buildSpec>
-	<natures>
-	</natures>
-</projectDescription>
diff --git a/Java/src/com/jameslow/AirPlay.java b/Java/src/com/jameslow/AirPlay.java
index d1cbd31..4c122fc 100755
--- a/Java/src/com/jameslow/AirPlay.java
+++ b/Java/src/com/jameslow/AirPlay.java
@@ -37,7 +37,8 @@ public class AirPlay {
 	protected int appletv_width = APPLETV_WIDTH;
 	protected int appletv_height = APPLETV_HEIGHT;
 	protected float appletv_aspect = APPLETV_ASPECT;
-	
+	protected boolean disableScaling;
+
 	//AirPlay class
 	public AirPlay(Service service) {
 		this(service.hostname,service.port,service.name);
@@ -58,6 +59,9 @@ public class AirPlay {
 		appletv_height = height;
 		appletv_aspect = (float) width/height;
 	}
+	public void setDisableScaling(boolean disableScaling) {
+		this.disableScaling = disableScaling;
+	}
 	public void setPassword(String password) {
 		this.password = password;
 	}
@@ -215,7 +219,7 @@ public class AirPlay {
 	protected BufferedImage scaleImage(BufferedImage image) {
 		int width = image.getWidth();
 		int height = image.getHeight();
-		if (width <= appletv_width && height <= appletv_height) {
+		if (disableScaling || width <= appletv_width && height <= appletv_height) {
 			return image;
 		} else {
 			int scaledheight;
@@ -493,7 +497,8 @@ public class AirPlay {
 	public static void selectResolutionDialog(Window parent, AirPlay airplay){
 		String[] choices = new String[]{
 				"Full HD  - 1080p - 1920x1080",
-				"HD Ready - 720p -1280 × 720"
+				"HD Ready - 720p -1280 × 720",
+				"No Scaling"
 				};
 
 		String input = (String) JOptionPane.showInputDialog(parent, "", "Select AppleTV Resolution",JOptionPane.PLAIN_MESSAGE, null, choices, choices[0]);
@@ -502,6 +507,8 @@ public class AirPlay {
 				airplay.setScreenSize(1920, 1080);
 			}else if(input.equals(choices[1])){
 				airplay.setScreenSize(1280, 720);
+			} else if(input.equals(choices[1])) {
+				airplay.setDisableScaling(true);
 			}
 		}
 	}
