--- startsmppsim.sh	2011-01-31 20:59:21.000000000 +1300
+++ startsmppsim.sh.new	2011-09-13 10:30:31.731548781 +1200
@@ -1,3 +1,6 @@
 # $Header: /var/cvsroot/SMPPSim2/startsmppsim.sh,v 1.6 2005/12/09 17:35:32 martin Exp $
 #! /bin/bash
-java -Djava.net.preferIPv4Stack=true -Djava.util.logging.config.file=conf/logging.properties -jar smppsim.jar conf/smppsim.props
\ No newline at end of file
+
+DIR=/opt/smpp-sim
+
+java -Djava.net.preferIPv4Stack=true -Djava.util.logging.config.file=$DIR/conf/logging.properties -jar $DIR/smppsim.jar $DIR/conf/smppsim.props
