diff -Naur kitchen_sync-0.52/src/ks.cpp kitchen_sync-0.52/src/ks.cpp
--- kitchen_sync-0.52/src/ks.cpp	2016-02-25 16:01:24.000000000 +1300
+++ kitchen_sync-0.52/src/ks.cpp	2016-06-15 15:07:38.769419380 +1200
@@ -46,7 +46,7 @@
 		if (options.from.password.empty()) options.from.password = "-";
 		if (options.set_from_variables.empty()) options.set_from_variables = "-";
 
-		const char *from_args[] = { ssh_binary.c_str(), "-C", "-c", "blowfish", options.via.c_str(),
+		const char *from_args[] = { ssh_binary.c_str(), "-C", "-c", "blowfish-cbc", options.via.c_str(),
 									from_binary.c_str(), "from", options.from.host.c_str(), options.from.port.c_str(), options.from.database.c_str(), options.from.username.c_str(), options.from.password.c_str(), options.set_from_variables.c_str(), options.filters.c_str(), nullptr };
 		const char **applicable_from_args = (options.via.empty() ? from_args + 5 : from_args);
 
