diff -Naur kitchen_sync-0.52.orig/CMakeLists.txt kitchen_sync-0.52/CMakeLists.txt
--- kitchen_sync-0.52.orig/CMakeLists.txt	2016-02-25 16:01:24.000000000 +1300
+++ kitchen_sync-0.52/CMakeLists.txt	2016-06-15 14:07:02.445401469 +1200
@@ -7,15 +7,11 @@
 include(CheckCXXCompilerFlag)
 CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
 CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
-CHECK_CXX_COMPILER_FLAG("-stdlib=libc++" COMPILER_SUPPORTS_STDLIB)
 if(COMPILER_SUPPORTS_CXX11)
 	ADD_DEFINITIONS("-std=c++11")
 elseif(COMPILER_SUPPORTS_CXX0X)
 	ADD_DEFINITIONS("-std=c++0x")
 endif()
-if(COMPILER_SUPPORTS_STDLIB)
-	ADD_DEFINITIONS("-stdlib=libc++")
-endif()
 
 # where to look first for cmake modules, before ${CMAKE_ROOT}/Modules/ is checked
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/Modules)
